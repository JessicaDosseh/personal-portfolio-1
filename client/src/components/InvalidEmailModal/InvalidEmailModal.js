import React from "react";
import PropTypes from "prop-types";
import styled, { css, keyframes } from "styled-components";

const buttonHover = keyframes`
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0.8;
  }
`;

const buttonHoverDark = keyframes`
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0.6;
  }
`;

const Modal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  padding: 0 5%;
  background: white;
  width: 100%;
  min-height: 100vh;
  z-index: 50;

  ${props =>
    props.darkmode &&
    css`
      background: #1b2331;
    `}

  div {
    height: 150px;
    max-width: 400px;
    width: 100%;
    margin: 30vh auto 0;
    z-index: 51;
    display: flex;
    flex-direction: column;
    background: none;

    p {
      width: 100%;
      text-align: center;
      color: black;
      margin-bottom: 35px;

      ${props =>
        props.darkmode &&
        css`
          color: white;
        `}
    }

    button {
      width: 100%;
      height: 45px;
      border: 0;
      border-radius: 5px;
      -moz-border-radius: 5px;
      -webkit-border-radius: 5px;
      background: #ad91ed;
      color: white;
      font-size: 1.6rem;
      text-transform: capitalize;
      cursor: pointer;

      ${props =>
        props.darkmode &&
        css`
          background: black;
        `}

      &:hover {
        animation: ${buttonHover} 0.2s forwards;

        ${props =>
          props.darkmode &&
          css`
            animation: ${buttonHoverDark} 0.2s forwards;
          `}
      }
    }
  }
`;

const InvalidEmailModal = ({ darkmode, setInvalidEmailModal }) => {
  const clickBtn = e => {
    e.preventDefault();
    setInvalidEmailModal(false);
  };
  return (
    <Modal darkmode={darkmode}>
      <div>
        <p>Enter a valid email.</p>
        <button onClick={clickBtn}>Okay</button>
      </div>
    </Modal>
  );
};

InvalidEmailModal.propTypes = {
  darkmode: PropTypes.bool,
  setInvalidEmailModal: PropTypes.func
};

export default InvalidEmailModal;
