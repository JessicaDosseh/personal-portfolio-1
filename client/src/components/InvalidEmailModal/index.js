export { default as InvalidEmailModal } from "./InvalidEmailModal";
