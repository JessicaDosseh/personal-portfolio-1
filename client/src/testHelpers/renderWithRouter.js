import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
