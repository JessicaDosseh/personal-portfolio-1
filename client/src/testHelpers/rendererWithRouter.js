import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
