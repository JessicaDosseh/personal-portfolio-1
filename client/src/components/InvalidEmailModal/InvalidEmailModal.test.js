import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import InvalidEmailModal from "./InvalidEmailModal";

describe("<InvalidEmailModal />", () => {
  it("should render the InvalidEmailModal component without crashing", () => {
    const testFn = jest.fn();
    const helpers = renderWithRouter(
      <InvalidEmailModal darkmode={false} setInvalidEmailModal={testFn} />
    );
  });

  it("should match the snapshot of InvalidEmailModal", () => {
    const tree = rendererWithRouter(
      <InvalidEmailModal darkmode={false} emailValues={{ success: true }} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
