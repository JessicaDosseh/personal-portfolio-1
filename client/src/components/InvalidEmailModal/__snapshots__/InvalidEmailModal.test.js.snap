// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvalidEmailModal /> should match the snapshot of InvalidEmailModal 1`] = `
<div
  className="sc-bdVaJa lnxrCL"
>
  <div>
    <p>
      Enter a valid email.
    </p>
    <button
      onClick={[Function]}
    >
      Okay
    </button>
  </div>
</div>
`;
